src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -139,6 +139,27 @@ public function map(callable $callback)
     }
 
     /**
+     * Reload a fresh model instance from the database for all the entities.
+     *
+     * @param  array|string  $with
+     * @return static
+     */
+    public function fresh($with = [])
+    {
+        $model = $this->first();
```

Comment:
Sorry I didn't see your comment and already did PR.
I don't need fresh on empty Collection personally but this check has to be there